src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -3793,9 +3793,70 @@ private static void NativeCompletionMemberName(string wordToComplete, List<Compl
             }
 
             CompleteMemberByInferredType(context, prevType, result, wordToComplete + "*", filter: IsPropertyMember, isStatic: false);
+            if (result.Count == 0)
+            {
+                // if we get here, we have not been able to infer the type to complete.
```

Comment:
Most of this code feels like it should be in it's own method because it might be useful elsewhere, basically everything up to the call to CompleteMemberByActualType.